### DISCUSSION TITLE
Make a semicolon optional

### DISCUSSION THREAD

**Discussion by IgorSivirinov** [dotnet/csharplang#4690] [Language Ideas] [answered]
### Summary
My idea is to make a semicolon optional
```C#
using System
using System.Security.Cryptography

namespace gRPCTestTiEventsy
{
    public class User
    {
        public string Email { get set }
        public string PasswordHash { get set }

        public User(string email, string password)
        {
            Email = email
            PasswordHash = HashPassword(password)
        }

        private static string HashPassword(string password)
        {
            byte[] salt
            byte[] buffer2

            if (password == null)
                throw new ArgumentNullException("password")
            
            using (var bytes = new Rfc2898DeriveBytes(password, 0x10, 0x3e8))
            {
                salt = bytes.Salt
                buffer2 = bytes.GetBytes(0x20)
            }

            var dst = new byte[0x31]

            Buffer.BlockCopy(salt, 0, dst, 1, 0x10)
            Buffer.BlockCopy(buffer2, 0, dst, 0x11, 0x20)

            return Convert.ToBase64String(dst)
        }

    }
}
```

### Motivation

Many languages (Kotlin, Go, Python, JavaScript) reject a semicolon.
What does that give us:
- Simplification of Syntax
- There will be no more comma-point error
- Users of C# coming from other languages will be easier to adapt to C#

**Comment by HaloFour** [comment, marked as answer]
No.

https://twitter.com/jaredpar/status/1010195445887336450

**Comment by jnm2** [comment]
Duplicate of https://github.com/dotnet/csharplang/discussions/496 and https://github.com/dotnet/roslyn/issues/5355.

**Comment by huoyaoyuan** [comment]
No semicolon will make syntax **much more complex**. Statement boudaries will require to be inferred, which can be not robust when little typo.
Making newline take part in the decision of statement boundary will also hurt the flexibility of code layout.
